opencog/rule-engine/Rule.cc

```diff
@@ -484,13 +484,28 @@ std::string Rule::to_string(const std::string& indent) const
 	return ss.str();
 }
 
+bool Rule::has_name_capture() const
+{
+	HandleSeq boundvars = (BindLinkCast(this->get_rule())->get_variables())
+			.varseq;
+	FreeVariables fv = (VariableListCast(this->get_vardecl()))
+			->get_variables();
+
+	for (const auto& v : boundvars)
+	{
+		if(fv.is_in_varset(v)) return true;
+	}
+	return false;
+}
```

Comment:
@SomeUserName1 I think this method always returns true. If `v` is in `varseq` it is gonna be in `varset.

Anyway, that does not solve the problem, what you want is just to make sure that the variables in bind link of `alpha_rule` https://github.com/opencog/atomspace/blob/master/opencog/rule-engine/Rule.cc#L407
are do not overlap with the variables in `vardecl`, the argument of `Rule::unify_source` https://github.com/opencog/atomspace/blob/master/opencog/rule-engine/Rule.cc#L396 that happens to be the variable declaration of `source`.

This is necessary so that subsequent unification properly works, otherwise it will assume that the variables with the same names are the same variables, while they are not.

If they are different then `rand_alpha_converted()` should rerun (or equivalently https://github.com/opencog/atomspace/blob/master/opencog/rule-engine/Rule.cc#L493 should be rerun) till all variables in the alpha converted rule are different from `vardecl` (the argument of `Rule::unify_source`).